### PR TITLE
Path: use java.util.Arrays convenience

### PR DESCRIPTION
**Pull request by jukzi**
instead of System.arraycopy

### CODE REVIEW

File: bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java [hunk 1]
```diff
@@ -510,8 +507,7 @@ private static String[] collapseParentReferences(boolean isAbsolute, String[] se
 		if (stackPointer == segmentCount)
 			return segments;
 		// build the new segment array backwards by popping the stack
-		String[] newSegments = new String[stackPointer];
-		System.arraycopy(stack, 0, newSegments, 0, stackPointer);
+		String[] newSegments = Arrays.copyOf(stack, stackPointer);
 		return newSegments;
```
Comment (HannesWell) [inline bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java:511]:
```suggestion
		return Arrays.copyOf(stack, stackPointer);
```

File: bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java [hunk 2]
```diff
@@ -1226,8 +1220,7 @@ public int segmentCount() {
 	 */
 	@Override
 	public String[] segments() {
-		String[] segmentCopy = new String[segments.length];
-		System.arraycopy(segments, 0, segmentCopy, 0, segments.length);
+		String[] segmentCopy = Arrays.copyOf(segments, segments.length);
 		return segmentCopy;
```
Comment (HannesWell) [inline bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java:1224]:
```suggestion
		return Arrays.copyOf(segments, segments.length);
```

**Review by HannesWell — COMMENTED** [review]
Looks good, thank you.
Only two calls can be further inlined.

File: bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java (hunk 2 above)
Comment (jukzi) [reply bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java]: done

**Review by HannesWell — APPROVED** [review]
Looks good.
Thank you.

**Pull request merged**